CefSharp.Example/DownloadHandler.cs

```diff
@@ -2,12 +2,23 @@
 //
 // Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
 
+using System;
+
 namespace CefSharp.Example
 {
     public class DownloadHandler : IDownloadHandler
     {
+        public event EventHandler<DownloadItem> OnBeforeDownloadFired;
+
+        public event EventHandler<DownloadItem> OnDownloadUpdatedFired;
+
         public void OnBeforeDownload(IBrowser browser, DownloadItem downloadItem, IBeforeDownloadCallback callback)
         {
+            if (this.OnBeforeDownloadFired != null)
```

Comment:
Would prefer firing of event handlers to look like below for consistency and best practice.

```c#
var handler = LoadingStateChanged;
if (handler != null)
{
    handler(this, args);
}
```

I know this is a demo, though I prefer to take a reference to a handler before executing as it's possible in a multithreaded environment for the handler reference to be cleared between the null check and executing. There's no